src/compiler/checker.ts

```diff
@@ -4715,10 +4718,11 @@ namespace ts {
             return true;
         }
 
-        // A valid base type is any non-generic object type or intersection of non-generic
+        // A valid base type is `any`, any non-generic object type or intersection of non-generic
         // object types.
         function isValidBaseType(type: Type): boolean {
-            return type.flags & (TypeFlags.Object | TypeFlags.NonPrimitive) && !isGenericMappedType(type) ||
+            return !!(type.flags & TypeFlags.Any) ||
+                type.flags & (TypeFlags.Object | TypeFlags.NonPrimitive) && !isGenericMappedType(type) ||

```

Comment:
I think it should work, although it's kind of a weird pun in that we rely on `!isGenericMappedType(anyType)` being true.